pkg/sources/git/git.go

```diff
@@ -1253,9 +1283,44 @@ func (s *Git) handleBinary(ctx context.Context, gitDir string, reporter sources.
 		return fmt.Errorf("error starting git cat-file: %w\n%s", err, stderr.Bytes())
 	}
 
-	defer func() { _ = cmd.Wait() }()
+	// Ensure all data from the reader (stdout) is consumed to prevent broken pipe errors.
+	// This operation discards any remaining data after HandleFile completion.
+	// If the reader is fully consumed, the copy is essentially a no-op.
+	// If an error occurs while discarding, it will be logged and combined with any existing error.
+	// The command's completion is then awaited and any execution errors are handled.
+	defer func() {
+		n, copyErr := io.Copy(io.Discard, stdout)
+		if copyErr != nil {
+			ctx.Logger().Error(
+				copyErr,
+				"Failed to discard remaining stdout data after HandleFile completion",
+			)
+		}
+
+		ctx.Logger().V(3).Info(
+			"HandleFile did not consume all stdout data; excess discarded",
+			"bytes_discarded", n)
+
+		err = combineErrors(err, fmt.Errorf("error discarding excess stdout data: %w", copyErr))
+
+		// Wait for the command to finish and handle any errors.
+		waitErr := cmd.Wait()
+		err = combineErrors(err, fmt.Errorf("command execution error: %w", waitErr))
+	}()
+
+	return handlers.HandleFile(catFileCtx, stdout, chunkSkel, reporter, handlers.WithSkipArchives(s.skipArchives))
+}
 
-	return handlers.HandleFile(ctx, stdout, chunkSkel, reporter, handlers.WithSkipArchives(s.skipArchives))
+// combineErrors combines two errors into one, preserving both error messages if they exist.
+func combineErrors(err1, err2 error) error {
+	switch {
+	case err1 != nil && err2 != nil:
+		return fmt.Errorf("%w; %w", err1, err2)
```

Comment:
Wait, you can do this? Which error is returned when you call `Unwrap()`?

Reply:
I thought you could do this—I was wrong. I can just go with `errors.Join`.

Reply:
Oh, nvm 🤣 Just read `In particular Unwrap does not unwrap errors returned by [Join](https://pkg.go.dev/errors#Join).`